File: examples/symmetry/src/proxy/server/proxy_server.c
#include <stddef.h>

typedef struct { unsigned long active_connections; unsigned long forwarded_requests; } proxy_server_t;
void proxy_server_init(proxy_server_t *s) { s->active_connections = 0; s->forwarded_requests = 0; }
void proxy_server_connect(proxy_server_t *s) { s->active_connections++; }
const char *proxy_server_forward(proxy_server_t *s, const char *target, const char *payload) { (void)target; s->forwarded_requests++; return payload; }
void proxy_server_disconnect(proxy_server_t *s) { if (s->active_connections) s->active_connections--; }

File: examples/symmetry/src/proxy/server/ProxyServer.java
package symmetry.proxy.server;

import java.util.concurrent.atomic.AtomicLong;

public final class ProxyServer {
    private final AtomicLong activeConnections = new AtomicLong();
    private final AtomicLong forwardedRequests = new AtomicLong();
    public void connect() { activeConnections.incrementAndGet(); }
    public String forward(String target, String payload) { forwardedRequests.incrementAndGet(); return payload; }
    public void disconnect() { activeConnections.decrementAndGet(); }
    public long getActiveConnections() { return activeConnections.get(); }
    public long getForwardedRequests() { return forwardedRequests.get(); }
}
